modules/core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/document/DocumentPropertiesJsonReader.java

```diff
@@ -200,14 +205,17 @@ private Object getScalarPropertyValue(Property property, JsonNode jn, Type type)
                         return blob.getString();
                     }
                 }
-                log.info("Unable to parse the property {}", property::getXPath);
-                return null;
+                throw new MarshallingException("Unable to parse the property " +  property.getXPath());
             }
             Object object = null;
             for (Class<?> clazz : resolver.getManagedClasses()) {
-                object = readEntity(clazz, clazz, jn);
-                if (object != null) {
-                    break;
+                try {
+                    object = readEntity(clazz, clazz, jn);
+                    if (object != null) {
+                        break;
+                    }
+                } catch (Exception e) {
```

Comment:
Shouldn't we want to handle `InterruptedException` and so on here? @efge?

But what about just catching `IOException` in fact?

Reply:
Yes I made a comment about this in the 10.10 PR as I was wondering if throwing instead of returning null was really backportable and if this was related too this `catch`. Any more specific exception than `Exception` would be better.

Reply:
We've changed the `Exception` type to catch to `MarshallingException` as it's all that we want..